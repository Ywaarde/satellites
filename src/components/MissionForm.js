import React from 'react';

export default class MissionForm extends React.Component {
  constructor(props){
    super(props);

    this.state = {
      MissionID: props.mission ? props.mission.mission_id : props.new_id,
      MissionNameShort: props.mission ? props.mission.name_short : '',
      MissionNameFull: props.mission ? props.mission.name_full : '',
      MissionAgencies: props.mission ? props.mission.agencies : '',
      MissionStatus: props.mission ? props.mission.status : ''
    };
  }

  onShortNameChange = (e) => {
    const MissionNameShort = e.target.value;
    this.setState({ MissionNameShort });
  }

  onFullNameChange = (e) => {
    const MissionNameFull = e.target.value;
    this.setState({ MissionNameFull });
  }

  onAgenciesChange = (e) => {
    const MissionAgencies = e.target.value;
    this.setState({ MissionAgencies });
  }

  onMissionStatusChange = (e) => {
    const MissionStatus = e.target.value;
    this.setState({ MissionStatus });
  }

  onSubmit = (e) => {
    e.preventDefault();
    this.props.onSubmit(this.state);
  }

  render() {
    return (
      <div>
        {this.props.mission && <h3>Mission ID {this.props.mission.mission_id}</h3>}
        {this.props.new_id && <h3>Mission ID {this.props.new_id}</h3>}
        <form onSubmit={this.onSubmit}>
          <input
            type='text'
            placeholder='Mission Short Name'
            required={true}
            autoFocus
            value={this.state.MissionNameShort}
            onChange={this.onShortNameChange}
          />
          <textarea
            placeholder='Mission Full Name'
            required={true}
            value={this.state.MissionNameFull}
            onChange={this.onFullNameChange}
          >
          </textarea>
          <input
            type='text'
            placeholder='Mission Agencies'
            required={true}
            value={this.state.MissionAgencies}
            onChange={this.onAgenciesChange}
          />
          <input
            type='text'
            placeholder='Mission Status'
            required={true}
            value={this.state.MissionStatus}
            onChange={this.onMissionStatusChange}
          />
          <button>Save Mission</button>
        </form>
      </div>
    );
  }
}

// Full mission details
    // Mission ID*
    // Mission Name Short*
    // Mission Name Full*
    // Mission Agencies*
    // Mission MissionStatus*
    // Launch Date,
    // EOL Date,
    // Applications,
    // Instruments,
    // Instrument IDs,
    // Orbit Type,
    // Orbit Altitude,
    // Orbit Period,
    // Orbit Inclination,
    // Repeat Cycle,
    // Orbit LST,
    // Orbit Longitude,
    // Orbit Sense,
    // Mission Site,
    // Data Access Portal
