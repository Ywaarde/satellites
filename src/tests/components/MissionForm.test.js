import React from 'react';
import { shallow } from 'enzyme';
import MissionForm from '../../components/MissionForm';
import testMissions from '../fixtures/missions';

test('Should render MissionList with mission', () => {
    const wrapper = shallow(<MissionForm mission={testMissions[0]}/>)
    expect(wrapper).toMatchSnapshot();
})

test('Should render MissionForm without missions', () => {
    const wrapper = shallow(<MissionForm />)
    expect(wrapper).toMatchSnapshot();
})

// test('should render error for invalid form submission', () => {
//     const wrapper = shallow(<MissionForm />)
//     wrapper.find('form').simulate('submit', {
//         preventDefault: () => {}
//     })
//     expect(wrapper.state('error').length).toBeGreaterThan(0)
// })

test('should set MissionNameShort on input change', () => {
    const value = 'Test input'
    const wrapper = shallow(<MissionForm />)
    expect(wrapper).toMatchSnapshot()
    wrapper.find('input').at(0).simulate('change', {
        target: { value } 
    })
    expect(wrapper.state('MissionNameShort')).toBe(value)
    expect(wrapper).toMatchSnapshot()
})

test('should set MissionNameFull on input change', () => {
    const value = 'Test input'
    const wrapper = shallow(<MissionForm />)
    expect(wrapper).toMatchSnapshot()
    wrapper.find('textarea').simulate('change', {
        target: { value } 
    })
    expect(wrapper.state('MissionNameFull')).toBe(value)
    expect(wrapper).toMatchSnapshot()
})

test('should set MissionAgencies on input change', () => {
    const value = 'Test input'
    const wrapper = shallow(<MissionForm />)
    wrapper.find('input').at(1).simulate('change', {
        target: { value } 
    })
    expect(wrapper.state('MissionAgencies')).toBe(value)
})

test('should set MissionStatus on input change', () => {
    const value = 'Test input'
    const wrapper = shallow(<MissionForm />)
    wrapper.find('input').at(2).simulate('change', {
        target: { value } 
    })
    expect(wrapper.state('MissionStatus')).toBe(value)
})

test('should call onSubmit prop for valid form submission', () => {
    const onSubmitSpy = jest.fn()
    const wrapper = shallow(<MissionForm mission={testMissions[0]} onSubmit={onSubmitSpy}/>)
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    })
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
        MissionID: testMissions[0].mission_id,
        MissionNameShort: testMissions[0].name_short,
        MissionNameFull: testMissions[0].name_full,
        MissionAgencies: testMissions[0].agencies,
        MissionStatus: testMissions[0].status
    })
})