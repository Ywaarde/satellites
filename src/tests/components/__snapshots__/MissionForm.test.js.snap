// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MissionForm without missions 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Mission Short Name"
      required={true}
      type="text"
      value=""
    />
    <textarea
      onChange={[Function]}
      placeholder="Mission Full Name"
      required={true}
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Mission Agencies"
      required={true}
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Mission Status"
      required={true}
      type="text"
      value=""
    />
    <button>
      Save Mission
    </button>
  </form>
</div>
`;

exports[`Should render MissionList with mission 1`] = `
<div>
  <h3>
    Mission ID 
    1
  </h3>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Mission Short Name"
      required={true}
      type="text"
      value="Mars"
    />
    <textarea
      onChange={[Function]}
      placeholder="Mission Full Name"
      required={true}
      value="Mission to Mars"
    />
    <input
      onChange={[Function]}
      placeholder="Mission Agencies"
      required={true}
      type="text"
      value="Fspace"
    />
    <input
      onChange={[Function]}
      placeholder="Mission Status"
      required={true}
      type="text"
      value="Approved"
    />
    <button>
      Save Mission
    </button>
  </form>
</div>
`;

exports[`should set MissionNameFull on input change 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Mission Short Name"
      required={true}
      type="text"
      value=""
    />
    <textarea
      onChange={[Function]}
      placeholder="Mission Full Name"
      required={true}
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Mission Agencies"
      required={true}
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Mission Status"
      required={true}
      type="text"
      value=""
    />
    <button>
      Save Mission
    </button>
  </form>
</div>
`;

exports[`should set MissionNameFull on input change 2`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Mission Short Name"
      required={true}
      type="text"
      value=""
    />
    <textarea
      onChange={[Function]}
      placeholder="Mission Full Name"
      required={true}
      value="Test input"
    />
    <input
      onChange={[Function]}
      placeholder="Mission Agencies"
      required={true}
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Mission Status"
      required={true}
      type="text"
      value=""
    />
    <button>
      Save Mission
    </button>
  </form>
</div>
`;

exports[`should set MissionNameShort on input change 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Mission Short Name"
      required={true}
      type="text"
      value=""
    />
    <textarea
      onChange={[Function]}
      placeholder="Mission Full Name"
      required={true}
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Mission Agencies"
      required={true}
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Mission Status"
      required={true}
      type="text"
      value=""
    />
    <button>
      Save Mission
    </button>
  </form>
</div>
`;

exports[`should set MissionNameShort on input change 2`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Mission Short Name"
      required={true}
      type="text"
      value="Test input"
    />
    <textarea
      onChange={[Function]}
      placeholder="Mission Full Name"
      required={true}
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Mission Agencies"
      required={true}
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Mission Status"
      required={true}
      type="text"
      value=""
    />
    <button>
      Save Mission
    </button>
  </form>
</div>
`;
